syntax = "proto3";

package streaming;

service FrameService {
  rpc Process(Query) returns (Response) {}
  rpc FindProcessed(ProcessedReq) returns (ProcessedResp) {}
}

enum QueryType {
  File = 0;
  Link = 1;
}

message Query {
  int64 id = 1;
  QueryType type = 2;
  string source = 3;
  optional int64 timeout = 4;
}

enum ResponseStatus {
  Success = 0;
  Processing = 1;
  Error = 2;
  Canceled = 3;
}

message Response {
  ResponseStatus status = 1;
}

message ProcessedReq {
  int64 queryId = 1;
}

message ProcessedResp {
  repeated string src = 1;
}